### PR TITLE
Add build + test GitHub actions.

### PR DESCRIPTION
**Pull request by pserwylo**
Now that we have test cases (thanks @juleskers), it seems silly not to have a CI job to run them for each commit/PR.

### CODE REVIEW

**Comment by pserwylo** [comment]
Not sure if it automatically starts running in this repo, or if it needs to be added in the settings somewhere. In my fork, it kind of just started running:

https://github.com/pserwylo/PianOli/actions/runs/6260859975/job/16999741173

**Comment by pserwylo** [comment]
I'm going to merge and just see if it starts to build.

**Comment by pserwylo** [comment]
Cool, it did just start to run: https://github.com/nicolasbrailo/PianOli/actions/runs/6260948984

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+
+  build-and-check:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - uses: actions/checkout@v2
```
Comment (juleskers) [inline .github/workflows/android.yml:17]: Seems all actions we use have later versions available, might as well use them to get the latest efficiency- and bug-fixes.

- [checkout@v4](https://github.com/actions/checkout)
- [setup-java@v3](https://github.com/actions/setup-java)
- [upload-artifact@v3](https://github.com/actions/upload-artifact)

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: set up JDK 17
+      uses: actions/setup-java@v1
+      with:
+        java-version: 17
+
+    - name: Gradle build
+      run: ./gradlew build
+
+    - name: Gradle check
+      run: ./gradlew check
```
Comment (juleskers) [inline .github/workflows/android.yml:28]: Oh, that's a lot simpler than I was reading! :astonished: 

All the blog posts I could find used obscure Action-definitions (obscure = <50 stars on marketplace, which can't be right for android-ecosystem-wide usage, so I was wondering what I was missing :smile: )

File: .github/workflows/android.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    - name: Archive build artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: outputs
+        path: app/build/outputs
```
Comment (juleskers) [inline .github/workflows/android.yml:34]: Maybe also the test-reports?
`app/build/reports`

**Review by juleskers — COMMENTED** [review]
That was fast! :blue_heart:

**Pull request merged**